src/main.rs

```diff
@@ -124,14 +137,144 @@ fn assert_no_uncommitted_changes(path: &PathBuf) -> Result<(), clap::error::Erro
+    // considered invalid.
+    if base_branch == "" {
+        return Err(Args::command().error(
+            clap::error::ErrorKind::ValueValidation,
+            format!("Cannot find a 'master' or 'main' base branch with which to compare the current branch '{}'of the spec", current_branch),
+        ));
+    }
+
+    // Finally, compute the diff between `current_branch` and `base_branch`.
+    // Return the diff so we can inform the rewrapper of which lines to format
+    // (as to avoid rewrapping the *entire* spec).
+    let git_diff = std::process::Command::new("git")
+        .arg("-C")
+        .arg(directory)
+        .arg("diff")
+        .arg("-U0")
+        .arg(base_branch)
+        .arg(current_branch)
+        .arg(filename_without_path)
+        .output()
+        .expect("Failed to compute `git diff`");
+
+    Ok(String::from_utf8(git_diff.stdout).unwrap())
+}
+
+// Takes the `String` output of `git_diff` above, and filters out irrelevant
+// lines. Cannot be a part of `git_diff` because this returns a vector of string
+// slices (for efficiency) on top of strings allocated inside of `git_diff`.
+fn sanitized_diff_lines(diff: &String) -> Vec<&str> {
+    diff.split("\n")
+        .enumerate()
+        // Strip the first 5 version control lines, and only consider lines
+        // prefixed with "+" that are more than one character long.
+        .filter(|&(i, line)| i > 4 && line.starts_with("+") && line.len() > 1)
+        // Remove the "+" version control prefix.
+        .map(|(_, line)| &line[1..])
+        .collect()
+}
+
+// Marks all of the lines in `lines` as needing format if and only if they
+// appear in `diff`. This algorithm is deficient in the sense that it compares
+// the *contents* of the lines in `diff` with `lines`, not the actual line
+// numbers. This is a problem if in the current branch of a spec, you add a line
+// that is identical to an earlier line in the spec that predates the current
+// branch. This algorithm will recognize the earlier, older line in the spec as
+// the one in the computed `diff`, and mark it as a candidate for formatting. In
+// practice, this is very unlikely to happen; it requires that:
+//   1.) A line you add is identical to a previous, preexisting one in the spec
+//   2.) The preexisting line in the spec appears after lines that exactly match
+//       all previous lines in the git diff.
+// Consider the example:
+//  +----------+
+// 1| My spec  |
+// 2|          |
+// 3|+ Intro   |
+// 4|+ My spec |
+//  +----------+
+// It might sound like line 1 would get marked as a candidate for reformatting
+// given the flaws in this algorithm, since it matches line 4 in the computed
+// diff, but it wouldn't. Only a line matching "My spec" that comees *after*
+// a line "Intro" would get marked as a candidate for reformatting.
+//
+// This problem would go away entirely once we give all lines in `diff` a proper
+// line number.
```

Comment:
I should probably file a bug for this and more-clearly describe the problem there...

Reply:
I've outlined this more clearly with tests and documentation, will file a bug later outlining this more.